Marlin/src/gcode/queue.cpp

```diff
@@ -467,11 +491,40 @@ void GCodeQueue::get_serial_commands() {
             if (n2pos) npos = n2pos;
           }
 
-          const long gcode_N = strtol(npos + 1, nullptr, 10);
-
+          const long gcode_N = strtol(npos + 1, nullptr, 10); // Host sent G-code Line# from RX_buffer
+
+          /**
+           * Resend Handler - Received line # != expected line number
+           *
+           * Standard behaviour - Clear RX_Buffer; resend request to host
+           * Extended behaviour - Allow ignore_resend_max # of errors to be ignored.
+           * Required when latency present between Marlin>>Host allowing G-code in-flight to cause comms issue
+           * on resend requests, such as stuttering and print failure.
+           * Ignore_resend_max should not exceed expected in-flight G-code + RX_Buffer. RX_Buffer is cleared in
+           * gcode_line_error, not ln_num_error_notice as next serial in buffer may be the required line.
+           */
           if (gcode_N != serial.last_N + 1 && !M110) {
             // In case of error on a serial port, don't prevent other serial port from making progress
-            gcode_line_error(F(STR_ERR_LINE_NO), p);
+            #if ENABLED(RESEND_HANDLER)
+              const serial_index_t serial_ind = p;
+
+              if ((resend_ctrl.ignore_resend_count < resend_ctrl.ignore_resend_max - 1)) {  // Threshold eliminated resends
```

Comment:
If `ignore_resend_max` (`RESEND_HANDLER_DROP_GCODE`) equals 1 then this test will never pass, since `ignore_resend_count < 0` can never be true. It seems like we would want this to pass once when `RESEND_HANDLER_DROP_GCODE` is set to 1, and then fail only after `ignore_resend_count` has been incremented once. See below for another point about `ignore_resend_count`….

Reply:
Correct. I was using an different numbering system and forgot to update this one. Remove '- 1' to resolve.